test/index.js

```diff
@@ -590,6 +590,11 @@ test('ap', function() {
   eq(Z.ap([inc], []), []);
   eq(Z.ap([inc], [1, 2, 3]), [2, 3, 4]);
   eq(Z.ap([inc, square], [1, 2, 3]), [2, 3, 4, 1, 4, 9]);
+  eq(Z.ap({}, {}), {});
+  eq(Z.ap({}, {x: 1, y: 2, z: 3}), {});
+  eq(Z.ap({x: inc}, {}), {});
+  eq(Z.ap({x: inc}, {x: 1}), {x: 2});
+  eq(Z.ap({x: inc, y: square}, {x: 1, y: 2}), {x: 2, y: 4});
```

Comment:
Let's include an assertion for the case in which the arguments do not have identical property names and the result is not `{}`.